pkg/scheduler/scheduler.go

```diff
@@ -588,4 +586,45 @@ func (sched *Scheduler) scheduleOne() {
 			metrics.PodScheduleSuccesses.Inc()
 		}
 	}()
+	return true
+}
+
+// scheduleOne does the entire scheduling workflow. if equivalence class feature is enabled,
+// Get the PodSet for the candidate pod and terate PodSet to do schedule one by one. Then,
+// delete the successfully scheduled pods from PodSet of this `Class`. If one pod failed to schedule,
+// the scheduler should return error, then the `Class` will be added to `unschedulableQ`. In this way,
+// we saved scheduling cycles for all the rest of pods in this `Class`.
+func (sched *Scheduler) scheduleOne() {
+	pod := sched.config.NextPod()
+	// pod could be nil when schedulerQueue is closed
+	if pod == nil {
+		return
+	}
+
+	if util.EquivalenceClassEnabled() {
+		var (
```

Comment:
IMO, we should move equiv and related codes to pkg/scheduler/core/generic_scheduler.go and keep Scheduler clean.